cypress/plugins/synapsedocker/index.ts

```diff
@@ -0,0 +1,169 @@
+/// <reference types="cypress" />
+
+import path from "path";
+import os from "os";
+import * as crypto from "crypto";
+import childProcess from "child_process";
+import * as fse from "fs-extra"
+
+// A cypress plugins to add command to start & stop synapses in
+// docker with preset templates.
+
+interface SynapseConfig {
+    configDir: string;
+    registrationSecret: string;
+}
+
+export interface SynapseInstance extends SynapseConfig {
+    synapseId: string;
+    port: number;
+}
+
+const synapses = new Map<string, SynapseInstance>();
+
+function randB64Bytes(numBytes: number): string {
+    return crypto.randomBytes(numBytes).toString("base64").replace(/=*$/, "");
+}
+
+async function cfgDirFromTemplate(template: string): Promise<SynapseConfig> {
+    const templateDir = path.join(__dirname, "templates", template);
+
+    const stats = await fse.stat(templateDir);
+    if (!stats || !stats.isDirectory) {
+        throw new Error(`No such template: ${template}`);
+    }
+    const tempDir = await fse.mkdtemp(path.join(os.tmpdir(), 'react-sdk-synapsedocker-'));
+
+    // change permissions on the temp directory so the docker container can see its contents
+    await fse.chmod(tempDir, 0o777);
```

Comment:
20% sure this won't work on windows, but also a good chance that nodejs will have stubbed it

Reply:
Nodejs says, "on Windows only the write permission can be changed, and the distinction among the permissions of group, owner or others is not implemented." so sounds like it will work to some extent, or at least no-op.